[2024-01-17T05:19:24.609+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Stream_Data_1.Reddit scheduled__2024-01-17T05:18:15.135468+00:00 [queued]>
[2024-01-17T05:19:24.619+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Stream_Data_1.Reddit scheduled__2024-01-17T05:18:15.135468+00:00 [queued]>
[2024-01-17T05:19:24.619+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2024-01-17T05:19:24.634+0000] {taskinstance.py:1382} INFO - Executing <Task(PythonOperator): Reddit> on 2024-01-17 05:18:15.135468+00:00
[2024-01-17T05:19:24.639+0000] {standard_task_runner.py:57} INFO - Started process 29993 to run task
[2024-01-17T05:19:24.643+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'Stream_Data_1', 'Reddit', 'scheduled__2024-01-17T05:18:15.135468+00:00', '--job-id', '415', '--raw', '--subdir', 'DAGS_FOLDER/kafka_stream.py', '--cfg-path', '/tmp/tmp1au8u3gh']
[2024-01-17T05:19:24.645+0000] {standard_task_runner.py:85} INFO - Job 415: Subtask Reddit
[2024-01-17T05:19:24.713+0000] {task_command.py:415} INFO - Running <TaskInstance: Stream_Data_1.Reddit scheduled__2024-01-17T05:18:15.135468+00:00 [running]> on host 614a419ad930
[2024-01-17T05:19:24.819+0000] {taskinstance.py:1660} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='me' AIRFLOW_CTX_DAG_ID='Stream_Data_1' AIRFLOW_CTX_TASK_ID='Reddit' AIRFLOW_CTX_EXECUTION_DATE='2024-01-17T05:18:15.135468+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2024-01-17T05:18:15.135468+00:00'
[2024-01-17T05:19:24.859+0000] {logging_mixin.py:151} INFO -  submission : 
[2024-01-17T05:19:25.735+0000] {logging_mixin.py:151} INFO - @True
[2024-01-17T05:19:25.737+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('192.168.224.8', 29092)]>: connecting to broker:29092 [('192.168.224.8', 29092) IPv4]
[2024-01-17T05:19:25.737+0000] {conn.py:1205} INFO - Probing node bootstrap-0 broker version
[2024-01-17T05:19:25.738+0000] {conn.py:418} ERROR - Connect attempt to <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('192.168.224.8', 29092)]> returned error 111. Disconnecting.
[2024-01-17T05:19:25.738+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('192.168.224.8', 29092)]>: Closing connection. KafkaConnectionError: 111 ECONNREFUSED
[2024-01-17T05:19:25.786+0000] {conn.py:380} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('192.168.224.8', 29092)]>: connecting to broker:29092 [('192.168.224.8', 29092) IPv4]
[2024-01-17T05:19:25.787+0000] {conn.py:418} ERROR - Connect attempt to <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('192.168.224.8', 29092)]> returned error 111. Disconnecting.
[2024-01-17T05:19:25.787+0000] {conn.py:919} INFO - <BrokerConnection node_id=bootstrap-0 host=broker:29092 <connecting> [IPv4 ('192.168.224.8', 29092)]>: Closing connection. KafkaConnectionError: 111 ECONNREFUSED
[2024-01-17T05:19:25.788+0000] {taskinstance.py:1943} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 192, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.9/site-packages/airflow/operators/python.py", line 209, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
  File "/opt/airflow/dags/kafka_stream.py", line 111, in reddit_function
    producer = KafkaProducer(bootstrap_servers=['broker:29092'], max_block_ms=5000)
  File "/home/airflow/.local/lib/python3.9/site-packages/kafka/producer/kafka.py", line 381, in __init__
    client = KafkaClient(metrics=self._metrics, metric_group_prefix='producer',
  File "/home/airflow/.local/lib/python3.9/site-packages/kafka/client_async.py", line 244, in __init__
    self.config['api_version'] = self.check_version(timeout=check_timeout)
  File "/home/airflow/.local/lib/python3.9/site-packages/kafka/client_async.py", line 900, in check_version
    raise Errors.NoBrokersAvailable()
kafka.errors.NoBrokersAvailable: NoBrokersAvailable
[2024-01-17T05:19:25.800+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=Stream_Data_1, task_id=Reddit, execution_date=20240117T051815, start_date=20240117T051924, end_date=20240117T051925
[2024-01-17T05:19:25.811+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 415 for task Reddit (NoBrokersAvailable; 29993)
[2024-01-17T05:19:25.820+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-01-17T05:19:25.839+0000] {taskinstance.py:2784} INFO - 0 downstream tasks scheduled from follow-on schedule check
